examples/wasm_custom_event.rs

```diff
@@ -0,0 +1,130 @@
+/// This example will show how to call a Rust function from JavaScript
+/// You should see the number 42 appear in browser console output when you click the button
+
+#[cfg(not(wasm_platform))]
+pub fn main() {
+    panic!("This example is only meant to be compiled for wasm target")
+}
+
+#[cfg(wasm_platform)]
+pub fn main() {
+    panic!("Please run `cargo run-wasm --example wasm_custom_event`")
+}
+
+#[cfg(wasm_platform)]
+mod wasm {
+    use std::cell::RefCell;
+
+    use wasm_bindgen::prelude::*;
+    use wasm_bindgen::JsCast;
+    use web_sys::HtmlScriptElement;
+    use winit::event::Event;
+    use winit::event::WindowEvent;
+    use winit::event_loop::EventLoop;
+    use winit::event_loop::EventLoopBuilder;
+    use winit::event_loop::EventLoopProxy;
+    use winit::window::Window;
+    use winit::window::WindowBuilder;
+
+    // Because EventLoopProxy is not Send, we need to wrap it in a RefCell and use thread_local!
+    thread_local! {
+        pub static EVENT_LOOP_PROXY: RefCell<Option<EventLoopProxy<CustomEvent>>> = RefCell::new(None);
+    }
```

Comment:
Since #2834 it is actually!
In any case it's still appropriate to use `thread_local!` if we aren't doing anything with web workers.
We should also probably use `once_cell::unsync::OnceCell` instead of `RefCell`.

Reply:
I am not certain want to make the comment now. Not super familiar with web workers. (js is single threaded unless you use web-workers is my understanding (?))